samples/cellular/modem_shell/src/location/location_shell.c

```diff
@@ -281,20 +357,21 @@ void location_ctrl_event_handler(const struct location_event_data *event_data)
 #if defined(CONFIG_LOCATION_DATA_DETAILS)
 		location_print_data_details(event_data->method, &event_data->location.details);
 #endif
-		if (gnss_location_to_cloud) {
-			gnss_location_work_data.loc_evt_data = *event_data;
-			gnss_location_work_data.format = gnss_location_to_cloud_format;
-			k_work_submit_to_queue(&mosh_common_work_q, &gnss_location_work_data.work);
+		if (event_data->method == LOCATION_METHOD_GNSS) {
+			location_cloud_work_data.send_cloud_gnss = arg_cloud_gnss;
 		}
 
 #if defined(CONFIG_DK_LIBRARY)
 		dk_set_led_on(LOCATION_STATUS_LED);
-		k_work_reschedule_for_queue(&mosh_common_work_q, &location_evt_led_work,
+		k_work_reschedule_for_queue(
```

Comment:
There can be work items which take longer time to execute in the `mosh_common_work_q`. Such work item could delay switching the LED off. This could be scheduled into the system work queue, which (ideally) should not be blocked for long.

Reply:
Changed to system work queue